components/server/src/authorization/authorizer.ts

```diff
@@ -116,40 +118,61 @@ export class Authorizer {
         );
     }
 
-    async hasPermissionOnProject(userId: string, permission: ProjectPermission, projectId: string): Promise<boolean> {
+    async hasPermissionOnProject(
+        userId: string,
+        permission: ProjectPermission,
+        project: { id: string; teamId?: string },
+    ): Promise<boolean> {
         if (userId === SYSTEM_USER) {
             return true;
         }
+        if (!project.teamId) {
+            return false;
+        }
 
         const req = v1.CheckPermissionRequest.create({
             subject: subject("user", userId),
             permission,
-            resource: object("project", projectId),
+            resource: object("project", project.id),
             consistency,
         });
 
-        return this.authorizer.check(req, { userId });
+        const result = await this.authorizer.check(project.teamId, req, { userId });
+        return result.permitted;
     }
 
-    async checkPermissionOnProject(userId: string, permission: ProjectPermission, projectId: string) {
-        if (await this.hasPermissionOnProject(userId, permission, projectId)) {
+    async checkPermissionOnProject(
+        userId: string,
+        permission: ProjectPermission,
+        project: { id: string; teamId?: string },
+    ) {
+        if (await this.hasPermissionOnProject(userId, permission, project)) {
             return;
         }
         // check if the user has read permission
-        if ("read_info" === permission || !(await this.hasPermissionOnProject(userId, "read_info", projectId))) {
-            throw new ApplicationError(ErrorCodes.NOT_FOUND, `Project ${projectId} not found.`);
+        if ("read_info" === permission || !(await this.hasPermissionOnProject(userId, "read_info", project))) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, `Project ${project.id} not found.`);
         }
 
         throw new ApplicationError(
             ErrorCodes.PERMISSION_DENIED,
-            `You do not have ${permission} on project ${projectId}`,
+            `You do not have ${permission} on project ${project.id}`,
         );
     }
 
-    async hasPermissionOnUser(userId: string, permission: UserPermission, resourceUserId: string): Promise<boolean> {
+    async hasPermissionOnUser(
+        userId: string,
+        permission: UserPermission,
+        resourceUserId: string,
+        resourceUser: Pick<User, "organizationId"> | undefined,
```

Comment:
lets make this explicit and pass in `organizationId: string`

Reply:
I'd much rather pass in `User`: it's much more intuitive, and less prone to errors. Also, we need to distinguish between "User with organizationId === undefined" and "no user passed (shape === undefined)". This way we keep the meaning + make it easy to use from the outside.